### PR TITLE
qlp_exactgrid

### PR DESCRIPTION
**Pull request by ryanli0070**
# submission name:
qlp_exactgrid

# upload zipped `archive.zip`
Hosted as a release asset (`curl -L` works):
https://github.com/ryanli0070/comma_video_compression_challenge/releases/download/qlp-exactgrid-v1/archive.zip

SHA-256 `869e666b4dd65d8da59f39a375505b65af28a695d94c73675e63864d678b8d1b`, 176,337 bytes, single ZIP member `x` (ZIP_STORED).

# report.txt
```
=== Evaluation results over 600 samples ===
  Average PoseNet Distortion: 0.00002902
  Average SegNet Distortion: 0.00056055
  Submission file size: 176,337 bytes
  Original uncompressed size: 37,545,489 bytes
  Compression Rate: 0.00469662
  Final score: 100*segnet_dist + √(10*posenet_dist) + 25*rate = 0.190506
```
(`evaluate.py` prints the score to 2 dp as `0.19`; full precision is `0.190506` on CPU. On the x86 `ubuntu-latest` runner it is `0.190503` — seg 0.00056052, pose 0.00002902.)

# does your submission require gpu for evaluation (inflation)?
no, inflate is CPU-only (device pinned to CPU); deps `numpy`, `torch`, `constriction`.

# did you include the compression script? and want it to be merged?
yes, `compress.sh` rebuilds `archive.zip` byte-for-byte from the polished latents + frozen decoder + searched selector; the polish/search pipeline is described below.

# is this submission competitive or innovative? explain why
Competitive. CPU score `0.190506` (x86 CI `0.190503`) vs the leaderboard's best (`rhnerv_comma` #112 `0.191126`, and the open PR #125 `hnerv_qlp` `0.190946`). It beats #125 on all three terms: SegNet ≈tie (`0.00056055` vs `0.00056085`), PoseNet better (`0.00002902` vs `0.00003000`), and archive smaller (`176,337` vs `176,525` B).

# additional comments
Frozen PR #95/#101 HNeRV decoder + PR #112 context range coder, both reused unchanged. Contribution: **exact-grid quantization-aware latent polish**. All 600×28 per-pair latents are optimized against the frozen SegNet/PoseNet through the *exact* inflate chain (bicubic→874×1164, #98 channel biases, straight-through clamp/round) with a boundary seg loss `sigmoid(-margin/τ)` (the smooth argmax-flip fraction, τ annealed) so gradient stays on the pixels that flip the metric. The key detail: the straight-through latent quantizer replicates the container's packing grid bit-for-bit (per-dim fp16 min/scale), so the polish the optimizer sees is exactly what ships (no train/pack gap). PR #101's 607-B sidecar is dropped (absorbed into the latents). A re-searched frame0 pose selector (concept from #110's FEC6, seg-neutral since SegNet sees only frame1) adds a small pose gain. Selection/eval are done in fp32 on CPU (fp16 selection overfits and regresses on the CPU axis). The latent-polish idea is concurrent with and shares its core with PR #125 `hnerv_qlp`; this was developed independently from that PR's public description. Full lineage/attribution in `submissions/qlp_exactgrid/THIRD_PARTY_NOTICES.md` (PRs #95/#98/#101/#110/#112, and #125).

### CODE REVIEW

**Comment by YassineYousfi** [comment]
added to leaderboard, congrats 
the code has too much overlap with code that is already merged though, so we won't merge it unless you refactor to reuse bits and pieces that are already there
feel free to reopen
